backend/security/input_validation.py

```
@@ -0,0 +1,294 @@
+        
+        # Check against allowed values
+        if field_name in ALLOWED_VALUES:
+            if str_value not in ALLOWED_VALUES[field_name]:
+                raise ValidationError(field_name, f"Value must be one of: {', '.join(ALLOWED_VALUES[field_name])}")
+        
+        # Type-specific validation
+        if field_name.endswith('_id') or field_name == 'id':
+            try:
+                return int(value)
+            except (ValueError, TypeError):
+                raise ValidationError(field_name, "Must be a valid integer")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)